internal/command/container.go

```diff
@@ -205,8 +206,15 @@ func goBuild(ctx Context, image containerImage) error {
 	}
 
 	// set output folder to fyne-cross/bin/<target>
-	binaryName := ctx.Name
+	binaryName := ctx.ExecutableName
+	if binaryName == "" {
+		binaryName = ctx.Name
+	}
 	if image.OS() == darwinOS {
+		if ctx.Engine.Name == kubernetesEngine && ctx.ExecutableName == "" {
+			return errors.New("executable name is required for darwinOS on kubernetes engine (use --executable-name)")
```

Comment:
I don't understand why this is required. Is there not a sane default we can use?

Reply:
The problem we have is that k8s can not be assumed to have access on the host to the source code, but it is trying to emulate what fyne build does which require access to source code.

The simple solution at this point is to bite the bullet and move fully to fyne build. It is clearly a technical debt problem and we are adding more and more code around it instead of solving it.